File: Ejercicios Corredor/Trabajo Corredor/Binario.cs
public class Binario {
    public byte[] Imagen = { 1, 0, 1, 0 };
    public byte[] Audio = { 10, 20, 30 };
    public byte[] Video = { 50, 60, 70 };
    public byte[] Archivo = { 65, 66, 67 };
    public byte[] Clave = { 1, 1, 0, 1 };

    public int ContarBytes(byte[] data) => data.Length;
    public byte ObtenerPrimerByte(byte[] data) => data[0];
    public bool EsVacio(byte[] data) => data.Length == 0;
    public byte SumarBytes(byte a, byte b) => (byte)(a + b);
    public byte[] Duplicar(byte[] data) => new byte[] { data[0], data[0] };
}

File: Ejercicios Corredor/Trabajo Corredor/Binario.go
package main
import "fmt"

type Binario struct {
	imagen  []byte
	audio   []byte
	video   []byte
	archivo []byte
	clave   []byte
}

func (b Binario) ContarBytes(data []byte) int { return len(data) }
func (b Binario) ObtenerPrimerByte(data []byte) byte { return data[0] }
func (b Binario) EsVacio(data []byte) bool { return len(data) == 0 }
func (b Binario) SumarBytes(a, c byte) byte { return a + c }
func (b Binario) Duplicar(data []byte) []byte { return []byte{data[0], data[0]} }

func main() {
	b := Binario{imagen: []byte{1, 0, 1, 0}}
	fmt.Println(b.ContarBytes(b.imagen))
}
